otherlibs/systhreads/st_stubs.c

```diff
@@ -265,13 +291,23 @@ static void caml_thread_enter_blocking_section(void)
 
 static void caml_thread_leave_blocking_section(void)
 {
-  caml_thread_t th = This_thread;
+  caml_thread_t th;
+#ifdef _WIN32
+  /* TlsGetValue calls SetLastError which will mask any error which occurred
```

Comment:
`TlsGetValue` doesn’t appear as such in the code, so the comment is rather cryptic.